src/sql/azdata.proposed.d.ts

```diff
@@ -1938,4 +1938,61 @@ declare module 'azdata' {
 			isPrimary: boolean;
 		}
 	}
+
+	export interface ChartData {
+		line: {
+			dataset: number[],
+			datasetLabel: string,
+			backgroundColor?: string;
+		};
+		doughnut: {
+			dataset: number[],
+			labels: string[],
+			colors?: string[];
+		};
+		bar: {
+			dataset: number[],
+			labels: string[],
+			datasetLabel: string,
+			colors?: string | string[];
+		};
+		horizontalBar: {
+			dataset: number[],
+			labels: string[],
+			datasetLabel: string,
+			colors?: string | string[];
+		};
+		pie: {
+			dataset: number[],
+			labels: string[],
+			colors?: string[];
+		};
+		radar: {
+			dataset: number[],
+			datasetLabel: string,
+			backgroundColor?: string;
+		};
+		polarArea: {
+			dataset: number[],
+			labels: string[],
+			colors?: string[];
+		};
+	}
+
+	export type ChartType = keyof ChartData;
+
+	export interface ModelBuilder {
+		chart<T extends ChartType>(): ComponentBuilder<ChartComponent<T>, ChartComponentProperties<T>>;
```

Comment:
Doing this lets users do weird stuff like `chart<'line' | 'bar'>`, which will then create a union of those two 

Is there a reason you need to have different properties per chart type? It seems like these are all generic properties that would apply to all charts, so it seems a lot more straightforward to have these properties all just be properties on `ChartComponentProperties` directly. And then the ChartType is just a string union 

`export type ChartType = 'line' | 'bar' | 'horizontalBar' | 'radar' | 'doughnut' | 'polarArea' | 'bubble' | 'pie' | 'scatter';`

(I also noticed you aren't including scatter or bubble, was that intentional?)

Reply:
We can add more chart specific properties/options to their respective interfaces.